coding/zlib.hpp

```diff
@@ -0,0 +1,126 @@
+#include "base/assert.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/string.hpp"
+
+#include "zlib.h"
+
+namespace coding
+{
+// Following class is a wrapper around ZLib routines.
+//
+// *NOTE* All Inflate() and Deflate() methods may return false in case
+// of errors. In this case the output sequence may be already
+// partially formed, so the user needs to implement its own roll-back
+// strategy.
+class ZLib
+{
+public:
+  enum class Level
+  {
+    NoCompression,
+    BestSpeed,
+    BestCompression,
+    DefaultCompression
+  };
+
+  template <typename OutIt>
+  static bool Deflate(char const * data, size_t size, Level level, OutIt out)
```

Comment:
Now there's only a single place where the code is used - tests. We're planning to use this later, and the practice will show us what signatures are useful and what aren't. From my point of view, the code will be used on vectors of chars and on strings, but, probably, it's a good idea to support iterators range too.

But your proposal is good, thanks. void* is more generic and more common for raw IO functions, Done.